### PR TITLE
gh-71759: Deprecate using bytes-like objects in builtins.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Deprecate using bytes-like objects except bytes and bytearray in
builtins int(), float(), compile(), eval() and exec().

* Issue: gh-71759

### CODE REVIEW

File: Doc/whatsnew/3.7.rst [hunk 1]
```diff
@@ -466,6 +466,10 @@ Deprecated
 
 - The :mod:`macpath` is now deprecated and will be removed in Python 3.8.
 
+- Deprecated using :term:`bytes-like objects <bytes-like object>` except
```
Comment (auvipy) [inline Doc/whatsnew/3.7.rst]: maybe this part should be moved to 3.8/rst?

Comment (serhiy-storchaka) [reply Doc/whatsnew/3.7.rst]: Indeed.

**Review by auvipy — APPROVED** [review]

**Comment by brettcannon** [comment]
@serhiy-storchaka did you still want to pursue this?

**Comment by serhiy-storchaka** [comment]
I am not sure. Removing support of general object providing the buffer protocol will simplify the code, but what if somebody needs it?

**Comment by brettcannon** [comment]
@serhiy-storchaka yeah, I don't have a good answer. I guess either ask on python-dev or close this PR to err on the side of backwards-compatibility.

**Review by iritkatriel — CHANGES_REQUESTED** [review]
This has merge conflicts now.

**Comment by iritkatriel** [comment]
Thanks @hauntsaninja .  

I think the discussion above between @serhiy-storchaka and @brettcannon is still unresolved. Not sure whether we will go ahead with this PR.

**Comment by encukou** [comment]
>  I guess either ask on python-dev or close this PR to err on the side of backwards-compatibility.

As the years go by, the current behaviour becomes more entrenched and backwards compatibility more important. Let's keep the status quo.

I'll close this PR next week if there are no objections.

**Pull request closed without merging**